desktop/src/com/unciv/app/desktop/UncivServer.kt

```diff
@@ -12,12 +12,25 @@ import java.io.File
 
 
 internal object UncivServer {
+    private var serverPort = 8080
+
     @JvmStatic
-    fun main(arg: Array<String>) {
+    fun main(args: Array<String>) {
+        args.forEach { arg ->
+            when {
+                arg.startsWith("-port=") -> with(arg.removePrefix("-port=").toIntOrNull() ?: 0) {
```

Comment:
I agree with @SomeTroglodyte that the default value for a non-parsable port should be 8080, which means instead of ?:-ing the result we can check it below (if this != null && this in 1024..49151)
However I think the current PR is useful enough to enter this version, and you can fix this minor issue in another insta-PR